docs: Update README.rst

YashNRam13 opened the pull request:
Updating broken URLs in the README.rst doc.

Fenn-CS reviewed (commented):
Looks okay, there are about 3+ broken links that needs to be addressed in https://tahoe-lafs.org/trac/tahoe-lafs/ticket/3749 I guess?

YashNRam13 commented:
> Looks okay, there are about 3+ broken links that needs to be addressed in https://tahoe-lafs.org/trac/tahoe-lafs/ticket/3749 I guess?

Hey Fenn, thanks for taking a look. Yes, there were three broken links. I've removed two completely (URLs to Windows and MacOS). We no longer have the MacOS article and we are redirecting users to the Windows stuff from the original Installation Doc.
I've updated the original installation doc article.

sajith reviewed (approved):
Looks good to me.  

I'm wondering if it makes sense to link to the corresponding rendered links in https://tahoe-lafs.readthedocs.io/en/latest/.   That way people who view the README from the web interface of code forges like GitHub can view the fully rendered document.

Or maybe not?  Some readers might be also viewing the README file locally, and they might be offline.

I guess that is a separate discussion, since there are several links like that in the README.

exarkun commented on newsfragments/3749.documentation:
```diff
@@ -0,0 +1 @@
+Fixes links to the installation and about Tahoe pages in the docs.
```
The style guide for news fragments is https://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles
